unittest/MutiThreadTest.h

```diff
@@ -0,0 +1,29 @@
+#pragma once
+
+#include <future>
+#include <vector>
+
+#include "knowhere/common/Dataset.h"
+#include "knowhere/index/VecIndex.h"
+
+namespace knowhere {
+
+namespace mthreadtest {
+std::vector<std::future<DatasetPtr>>&
+GetFeatureQueue() {
+    static std::vector<std::future<DatasetPtr>> qu;
+    return qu;
+}
+
+DatasetPtr
+Async(VecIndexPtr& idx, DatasetPtr& dataset, Config& cfg, faiss::BitsetView bitset) {
+    std::vector<std::future<DatasetPtr>>& qu = GetFeatureQueue();
```

Comment:
Can I ask why we need this?

Reply:
To store the results, it seems like the test team do the concurrent operations, and the results were obtained.